File: src/main/java/_01_imperative_style/Sample.java
package _01_imperative_style;

import java.util.List;

import static util.AirportNameUtil.getNameOfAirport;

public class Sample {

    public static void main(String[] args) {
        List<String> iataCodes = List.of("AUS", "IAH", "DFW", "TAS", "SAT");

        for(var iataCode: iataCodes) {
            try {
                System.out.println(getNameOfAirport(iataCode));
            } catch(Exception ex) {
                System.out.println(ex.getMessage());
            }
        }
    }
}

File: src/main/kotlin/_08_functional_deal_with_it_kotlin/Sample.kt
package _08_functional_deal_with_it_kotlin

import io.vavr.control.Try
import util.AirportNameUtil.getNameOfAirport

fun main() {
    val iataCodes = listOf("AUS", "IAH", "DFW", "TAS", "SAT")

    // If an exception occurs, an error message is displayed along with "Error: "
    iataCodes
        .map { Try.of { getNameOfAirport(it) } }
        .map { it.map { airportName -> airportName.uppercase() } }
        .map { tryName -> tryName.recover { ex -> "Error: ${ex.message}" }.get() }
        .forEach { println(it) }

    println("===============================")

    // If an exception occurs, ignore the iataCode
    iataCodes
        .map { Try.of { getNameOfAirport(it) } }
        .map { it.map { airportName -> airportName.uppercase() } }
        .filter { it.isSuccess }
        .map { it.get() }
        .forEach { println(it) }
}
